src/gui/main.cpp

```diff
@@ -51,9 +51,10 @@ int cli_main(int argc, char **argv)
 		argsNoFork << argv[i];
 	}
 
-	NeovimQt::App app(argc, argv);
-
-	app.checkArgumentsMayTerminate();
+	QCoreApplication app(argc, argv);
+	QCommandLineParser p;
+	NeovimQt::App::processCommandlineOptions(p, app.arguments());
+	NeovimQt::App::checkArgumentsMayTerminate(p);
```

Comment:
I''m not entirely happy with this just yet. But i cant fully split cli argument handling from App.